apps/pie-microsite/src/_11ty/shortcodes/tokensTable/tokenTypes/elevation.js

```diff
@@ -0,0 +1,65 @@
+const { deindentHTML } = require('../../shortcode-utilities');
+
+/**
+ * Builds the example elevation box-shadows to show on the token list item
+ * @param {string} token - the token value
+ * @param {*} tokenMetadata - the metadata for the token. data such as descriptions
+ * @param {path} path - the categorised token path (light or dark)
+ * @returns {string} - the elevation swatch example HTML string
+ */
+const buildElevationExample = (token, tokenMetadata, path) => {
+    const boxShadowValues = token.shadows.map(({
+        x,
+        y,
+        blur,
+        spread,
+        r,
+        g,
+        b,
+        opacity
+    }) => `${x}px ${y}px ${blur}px ${spread}px rgba(${r}, ${g}, ${b}, ${opacity})`);
+
+    const isDefaultElevation = tokenMetadata.category === 'defaultElevation';
+    const theme = path.includes('light') ? 'light' : 'dark';
+
+    const styling = {
+        light: {
+            container: isDefaultElevation ? '#f5f3f1' : '#3B3A39',
+            elevationBox: isDefaultElevation ? '#fff' : '#575655'
+        },
+        dark: {
+            container: isDefaultElevation ? '#3B3A39' : '#f5f3f1',
+            elevationBox: isDefaultElevation ? '#575655' : '#fff'
+        }
+    };
+
+    const container = `--example-container: ${styling[theme].container};`;
+    const elevationBox = `--example-elevation: ${styling[theme].elevationBox};`;
+    const elevationBoxShadow = `--example-shadow: ${boxShadowValues};`;
+
+    return (`<div class="c-tokensTable-example-container--elevation" style="${container}"><div class="c-tokensTable-example--elevation" style="${elevationBox} ${elevationBoxShadow}"></div></></div>`);
+};
+
+/**
+ * Builds the overall token description element for global and alias Elevation tokens.
+ * @param {*} tokenMetadata the metadata for the token. data such as descriptions
+ * @returns {string} - the description HTML string
+ */
+const buildElevationDescription = tokenMetadata => {
```

Comment:
Is this the same function that colour uses? And is the only thing it does differently to the default description adding the `u-spacing-b--bottom`.

I'm not sure if it makes it more or less confusing 😂 , but instead of having custom functions for elevation and colour is it worth having a `shouldHaveBottomSpacing` variable that adds the class on instead?

Reply:
ooh yeah good spot, can probably move to the main tokensTable index.js instead! Briefly spoke about this earlier and actually I think the spacing is supposed to be the same for all token types, so we probably won't even need the `shouldHaveBottomSpacing` variable 😄

Reply:
I think the logic of this function is the same for the spacing description as well, so it makes sense to generalize it and remove the tokenTypeBuilder logic for now. wdyt?

Reply:
yep as @raoufswe says, lets bring that logic out of colour and into a generic description builder. Given we're hard coding all descriptions into JSON basically the same way as colour there's no reason for token-specific handlers anymore 👍

Reply:
@FayeCarter @LTurns they will all now need the `u-spacing-b--bottom` class :)

Reply:
🥳